File: src/main/java/fi/quanfoxes/lexer/ActionOperator.java
package fi.quanfoxes.lexer;

public class ActionOperator extends Operator {
    private Operator operator;

    public ActionOperator(String identifier, Operator operator, int priority) {
        super(identifier, OperatorType.ACTION, priority);
        this.operator = operator;
    }

    /**
     * Returns the operation associated with the action
     * @return Operation associated with the action
     */
    public Operator getOperator() {
        return operator;
    }
}

File: Zigzag/Lexer/ActionOperator.cs
using System;
using System.Collections.Generic;

public class ActionOperator : Operator
{
	public Operator? Operator { get; private set; }

	public ActionOperator(string identifier, Operator? @operator, int priority) : base(identifier, OperatorType.ACTION, priority)
	{
		Operator = @operator;
	}

	public override bool Equals(object? obj)
	{
		return obj is ActionOperator @operator &&
			   base.Equals(obj) &&
			   EqualityComparer<Operator>.Default.Equals(Operator, @operator.Operator);
	}

	public override int GetHashCode()
	{
		return HashCode.Combine(base.GetHashCode(), Operator);
	}
}